atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/transaction/Transactions2TableInteraction.java

```diff
@@ -0,0 +1,149 @@
+                startTimestamp, Bytes.getArray(row.getBytes(CassandraConstants.VALUE)));
+        return new TransactionTableEntry(startTimestamp, Optional.of(commitTimestamp));
+    }
+
+    @Override
+    public Statement bindCheckStatement(PreparedStatement preparedCheckStatement, long startTs, long _commitTs) {
+        Cell cell = TicketsEncodingStrategy.INSTANCE.encodeStartTimestampAsCell(startTs);
+        ByteBuffer rowKeyBb = ByteBuffer.wrap(cell.getRowName());
+        ByteBuffer columnNameBb = ByteBuffer.wrap(cell.getColumnName());
+        BoundStatement bound = preparedCheckStatement.bind(rowKeyBb, columnNameBb);
+        return bound.setConsistencyLevel(ConsistencyLevel.QUORUM)
+                .setSerialConsistencyLevel(ConsistencyLevel.SERIAL)
+                .setReadTimeoutMillis(LONG_READ_TIMEOUT_MS)
+                .setRetryPolicy(DefaultRetryPolicy.INSTANCE);
+    }
+
+    @Override
+    public Statement bindAbortStatement(PreparedStatement preparedAbortStatement, long startTs, long commitTs) {
+        Cell cell = TicketsEncodingStrategy.INSTANCE.encodeStartTimestampAsCell(startTs);
+        ByteBuffer rowKeyBb = ByteBuffer.wrap(cell.getRowName());
+        ByteBuffer columnNameBb = ByteBuffer.wrap(cell.getColumnName());
+        ByteBuffer valueBb =
+                ByteBuffer.wrap(TicketsEncodingStrategy.INSTANCE.encodeCommitTimestampAsValue(startTs, commitTs));
+        BoundStatement bound = preparedAbortStatement.bind(rowKeyBb, columnNameBb, valueBb);
+        return bound.setConsistencyLevel(ConsistencyLevel.QUORUM)
+                .setSerialConsistencyLevel(ConsistencyLevel.SERIAL)
+                .setDefaultTimestamp(CassandraConstants.CAS_TABLE_TIMESTAMP)
+                .setReadTimeoutMillis(LONG_READ_TIMEOUT_MS)
+                .setIdempotent(true) // by default CAS operations are not idempotent in case of multiple clients
+                .setRetryPolicy(abortRetryPolicy);
+    }
+
+    @Override
+    public boolean isRowAbortedTransaction(Row row) {
+        return Arrays.equals(
+                Bytes.getArray(row.getBytes(CassandraConstants.VALUE)),
+                TicketsEncodingStrategy.ABORTED_TRANSACTION_VALUE);
+    }
+
+    @Override
+    public List<Statement> createSelectStatements(TableMetadata transactionsTable) {
+        Set<ByteBuffer> encodedRowKeys = TicketsEncodingStrategy.INSTANCE
+                .encodeRangeOfStartTimestampsAsRows(
+                        timestampRange.inclusiveLowerBound(), timestampRange.inclusiveUpperBound())
+                .map(ByteBuffer::wrap)
+                .collect(Collectors.toSet());
+        return encodedRowKeys.stream()
+                .map(rowKey -> QueryBuilder.select()
+                        .all()
+                        .from(transactionsTable)
+                        .where(QueryBuilder.eq(CassandraConstants.ROW, rowKey))
+                        .setConsistencyLevel(ConsistencyLevel.ALL)
```

Comment:
We should document why consistency ALL is required (it's the main change I'm seeing here from the implementation in internal-backup service). It's to better flush out partially applied CASes, but we should explain what that is.